pkg/index/tree.go

```diff
@@ -593,12 +593,16 @@ func (n *Node[V]) InsertAfter(newNode, referenceNode *Node[V]) error {
 	return nil
 }
 
-// HasTextChild returns true if the node has a text child.
+// HasTextChild returns true if the node's children consist of only text children.
 func (n *Node[V]) HasTextChild() bool {
-	for _, child := range n.Children() {
-		if child.IsText() {
-			return true
+	if len(n.children) > 0 {
```

Comment:
`n.children` does not filter out tombstone nodes. Is this the correct conditional expression?

Reply:
No, it wasn't intended, and it's not a correct conditional expression. I will fix it.